h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -41,43 +41,101 @@ class SearchBarController extends Controller {
       }.bind(this), 0);
     };
 
-    var getPreviousSiblingElement = element => {
+    var isVisible = element => {
+      return element &&
+        (element.nodeType !== 1 ||
+          !element.classList ||
+          element.classList.contains('is-hidden'));
+    };
+
+    var getPreviousVisibleSiblingElement = element => {
       if (!element) {
         return null;
       }
 
       do {
         element = element.previousSibling;
-      } while (element && element.nodeType != 1);
-
+      } while (isVisible(element));
       return element;
     };
 
-    var getNextSiblingElement = element => {
+    var getNextVisibleSiblingElement = element => {
       if (!element) {
         return null;
       }
 
       do {
         element = element.nextSibling;
-      } while (element && element.nodeType != 1);
+      } while (isVisible(element));
 
       return element;
     };
 
+    var unHideDropdownItems = () => {
+      // Reset dropdown items to not hidden
+      this._dropdownItems.forEach(function(dropdownItem) {
+        dropdownItem.classList.remove('is-hidden');
+      });
+    };
+
     var closeDropdown = () => {
       clearActiveDropdownItem();
+      unHideDropdownItems();
       this.setState({open: false});
       this._input.removeEventListener('keydown', setupListenerKeys,
         true /*capture*/);
     };
 
     var openDropdown = () => {
+      clearActiveDropdownItem();
+
       this.setState({open: true});
+
       this._input.addEventListener('keydown', setupListenerKeys,
         true /*capture*/);
     };
 
+    var getVisibleDropdownItems = () => {
+      return this._dropdown.querySelectorAll('li:not(.is-hidden)');
+    };
+
+    var setVisibleDropdownItems = word => {
+      // Show items that match the word and ones that don't
```

Comment:
Typo, I think this comment should read: `// Show items that match the word and hide ones that don't.` Also it might be worth using a [jsdoc](http://usejsdoc.org/) comment here? We've got those in other js code.